src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -79,5 +79,28 @@ public static async Task ReadPrimitivesWithTrailingTriviaAsync()
                 Assert.Equal(1, i);
             }
         }
+
+        [Fact]
+        public static async Task TestBOMWithSingleJsonValue()
+        {
+            byte[] utf8BomAndValueArray = new byte[] { 0xEF, 0xBB, 0xBF, 49 };
+            byte value;
+            using (Stream stream = new MemoryStream(utf8BomAndValueArray))
+            {
+                value = await JsonSerializer.ReadAsync<byte>(stream);
+            }
+            Assert.Equal(1, value);
+
+            using (Stream stream = new MemoryStream(utf8BomAndValueArray))
+            {
+                JsonSerializerOptions options = new JsonSerializerOptions
+                {
+                    DefaultBufferSize = 1,
```

Comment:
nit: Consider separating this out to a separate `[Theory]` test with a few `DefaultBufferSize` values (1, 2, 3, 4, 15, 16, 17)

We may also want to have JSON data that is both under and above the default buffer size in each of these cases.

So, in the case of `DefaultBufferSize` = 17, maybe have the input json be "UTF-8 BOM followed by an array of numbers" such that the length is > 17 along with the one where the input json with just the bom plus a single value.